scripts/linters/pylint_extensions.py

```diff
@@ -1484,6 +1484,16 @@ class RestrictedImportChecker(checkers.BaseChecker):
             'domain layer and controller layer respectively.'),
     }
 
+    # Mapping between modules and imports that are forbidden in these modules.
+    # The key part needs to be absolute path to the module for which we want to
+    # forbid the imports. Both import formats ('import xyz [as klm]' and
+    # 'from xyz import abc [as klm]') are checked.
+    module_to_forbidden_imports = {
+        'oppia.core.storage': ['core.domain'],
+        'oppia.core.domain': ['core.controllers'],
+        'oppia.core.controllers': ['core.platform', 'core.storage']
+    }
```

Comment:
Can you please check whether we can move this to .pylintrc? (We are structuring the pylint_extensions files such that it doesn't depend on Oppia codebase structure/terms.) [You can check [existing structure](https://github.com/oppia/oppia/blob/develop/scripts/linters/pylint_extensions.py#L549) on how to provide options in checkers.]

Reply:
Hmmm, I was looking into how options work in pylint. The most complicated structure provided is lsit of items so it is not possible to encode this structure in one param. It would be possible to have an option for each folder that we have and provide a list of forbidden imports for that list.

In .pylintrc it would look like
```
forbidden-imports-core-storage=core.domain
forbidden-imports-core-domain=core.controllers
forbidden-imports-core-controllers=core.platform, core.storage
```

what do you think?

Reply:
Can we have a patter for this:

`forbidden-imports=(module1=>moule2|module3, module6=>moule5|module4)`

We will assume it as module1 should not be imported in module2 and module3?
You can have a parser in the check which will parse the csv strings
```
mod_paths = csv[0].split('=>')
assert mod_paths == 2

diss_allow_import = mod_path[0]
for_modules = mod_path[1].splt('|')
```
(not sure about the variable names)
You can check PR #11557 for ref*